### PR TITLE
Feature/#32 option menu

### LINKED ISSUE #32: Option menu
as mentioned in #28

### PR DESCRIPTION
**Pull request by Serraniel**
- [x] Firefox is missing

But the rest of the changes is ready for review I think.

### CODE REVIEW

**Comment by Serraniel** [comment]
I added an implementation for firefox but it´s a bit of a nightmare to validate it, as you cannot use the sync storage with a temporary addon. It will give the following error:
> Uncaught (in promise) Error: The storage API will not work with a temporary addon ID. Please add an explicit addon ID to your manifest. For more information see https://bugzil.la/1323228.

However, I validated the code itself is working by just using the local storage instead of sync storage. 

Additionally I found out, that the chrome local storage api also is working in firefox. Just as a side fact.

**Comment by Serraniel** [comment]
I think that´s it. Also decided to not open settings in tab, so the browser show that kind of overlay which should be enough imo and saves us from further styiling. 

201108_13-13-22-518 Erweiterungen_-_Aniwatch_Plus_-_Google_Chrome201108_13-13-35-339 Erweiterungen_-_Aniwatch_Plus_-_Google_Chrome

File: src/javascript/configuration/configuration.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    constructor() {
+        this.settingsCache = new Map();
+    }
+
+    getProperty(key, callback) {
+        if (this.settingsCache.has(key)) {
+            callback(this.settingsCache.get(key));
+        }
+        else {
+            // OOOPS // currently all settings are default true. This isn´t a problem but there should be much better soloutions after migration to typescript....
```
Comment (Serraniel) [inline src/javascript/configuration/configuration.js:27]: This should be refactored and improved after #48 .

File: src/javascript/enhancements/languageDisplay.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 import * as helper from '../utils/helpers';
 
 export function init() {
-    core.registerScript(node => {
-        // run the scripts
-        if (helper.isHtmlElement(node)) {
-            updateLanguageDisplay(node)
+    getGlobalConfiguration().getProperty(SETTINGS_animeLanguageDisplay, value => {
+        if (value) {
+            if (getGlobalConfiguration().animeLanguageDisplay) {
```
Comment (Serraniel) [inline src/javascript/enhancements/languageDisplay.js]: That if statement has to be removed.

Comment (Serraniel) [reply src/javascript/enhancements/languageDisplay.js]: Done and working, agaub.

**Review by kaffem — APPROVED** [review]
Looks good!

**Pull request merged**